pkg/admission/customresourcevalidation/attributes.go

```diff
@@ -0,0 +1,46 @@
+package customresourcevalidation
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apiserver/pkg/admission"
+)
+
+// unstructuredUnpackingAttributes tries to convert to a real object in the config scheme
+type unstructuredUnpackingAttributes struct {
+	admission.Attributes
+}
+
+func (a *unstructuredUnpackingAttributes) GetObject() runtime.Object {
+	return toBestObjectPossible(a.Attributes.GetObject())
+}
+
+func (a *unstructuredUnpackingAttributes) GetOldObject() runtime.Object {
+	return toBestObjectPossible(a.Attributes.GetOldObject())
+}
+
+// toBestObjectPossible tries to convert to a real object in the config scheme
+func toBestObjectPossible(orig runtime.Object) runtime.Object {
```

Comment:
This function really could just be a shared helper that takes the unstructured and the into and then does the type check (verify the config scheme allows `into`, then run FromUnstructured on it).  That bypasses the cast and make this much easier to read.

```
var image *configv1.Image
if errs := ConvertFromUnstructured(obj, image); len(errs) > 0 {
  return errs
}
...
```

instead of having to do the very ugly cast.

Reply:
> This function really could just be a shared helper that takes the unstructured and the into

The into isn't known, that's part of what this function is figuring out.  If it gets specified then I have reflection trying to clone it.  Also the unstructured has to be cast somewhere.